### ISSUE TITLE
Is it possible to hide metadata on shared links?

### ISSUE THREAD

**Issue by Bart1909** [photoprism/photoprism#406]
When using shared links, some metadata like date, file size, dimensions, camera model etc. are shown.

Would be nice, if this could be configured / disabled.

**Comment by lastzero** [comment]
Thought about this too, but needs more discussion... What exactly should be hidden? And why? Typical use cases?

Maybe we can offer use case specific presets instead of adding complicated config options for every single field...

The date for example is very important and should typically not be removed, unless maybe it's some sort of professional showcase and not a personal album you share with friends.

**Comment by Bart1909** [comment]
My use case will be a gallery for a Photo Booth.
So all photos in one folder / album will have the same date.
Because we only want to share the photos with the people, none of the meta data are required to be shown.

**Comment by lastzero** [comment]
So it only needs to be hidden in the photo viewer (lightbox) and not on the API level?

**Comment by Bart1909** [comment]
Yes, and only in the photo viewer for shared links / guests (if this will be easier)

**Comment by lastzero** [comment]
If your concern is security or privacy, then just hiding metadata in the UI doesn't ultimately solve the issue.

We'll add more config options (and templates, also for sharing) after our release. Right now, we only add / change what seems absolutely necessary.

**Comment by Bart1909** [comment]
The concern is not really security.
But the user does not need this information, so it can be hidden and makes the page "lighter".
If it's retrievable via api that's fine for our use case

**Comment by lastzero** [comment]
At least on mobile devices / touch screens, you can easily hide all metadata by quickly tipping on the screen. Also during slideshows.

![Screenshot_20200715_130102_com android chrome](https://user-images.githubusercontent.com/301686/87537791-7f86e980-c69b-11ea-95d7-239abda8657d.jpg)

**Comment by graciousgrey** [comment]
> At least on mobile devices / touch screens, you can easily hide all metadata by quickly tipping on the screen. Also during slideshows.

@Bart1909 Does this already solve the issue?

**Comment by kevincox** [comment]
An huge feature here is hiding location. Most phones tag photos with location metadata which is super useful when browsing my own library. But I would love to share an album of cat photos without revealing where I live. 

I think it would be great if there was an option to restrict metadata on sharing links. For example when creating a link there would be an option "Full Metadata" like current or you can flip it off in which case all metadata is kept secret. So the only data available would be pixel data (and required basic info like file size, dimensions). Then various switches could be available for exposing more data such as capture time, hardware information, location metadata and anything else that someone has a use case for.

All stripped metadata must be completely blocked. Including in the API and stripped from any downloaded images (and thumbnails if they contain this info right now).

**Comment by dror3go** [comment]
This is a feature I'm also looking for.
I also isn't too happy about adding more config options, and I think @kevincox 's idea for share-specific settings is a great solution for this 👍
